blueprints/banking_assistant/__init__.py

```diff
@@ -0,0 +1,498 @@
+# -*- coding: utf-8 -*-
+"""This module contains a MindMeld application"""
+from mindmeld import Application
+from mindmeld.core import FormEntity
+import random
+import json
+import os
+
+app = Application(__name__)
+
+__all__ = ["app"]
+
+# User data class and helper functions for handlers
+
+user = None
+
+
+class User:
+    def __init__(self, request):
+        self.sample_users = {
+            "johndoe123": 0,
+            "larry_l12": 1,
+            "splashbro30": 2,
+        }
+        user = self._pull_data(request)
+        directory = os.path.dirname(os.path.abspath(__file__))
+        path = os.path.join(directory, "data/sample_user_data.json")
+        with open(path) as f:
+            data = json.load(f)
+            self.data = data[user]
+
+    def _pull_data(self, request):
+        if request.context.get("user_name"):
+            user_name = request.context.get("user_name")
+            self.user = self.sample_users[user_name]
+        else:
+            self.user = random.choice([0, 1, 2])
+        return self.user
+
+    def put(self, key, value):
+        self.data[key] = value
+
+    def get(self, key):
+        return self.data[key]
+
+    def check_value(self, responder):
+        return self.get(responder.slots["origin"]) >= responder.slots["amount"]
+
+
+def _credit_amount_helper(request, responder, entity, user):
+    responder.slots["payment"] = (
+        entity["value"][0]["cname"] if len(entity["value"]) > 0 else entity["text"]
+    )
+    if responder.slots["payment"] == "minimum":
+        responder.reply(
+            "OK, we have scheduled your credit card payment for your minimum balance of $"
+            "{min:.2f}"
+        )
+        user.put("credit", user.get("credit") - responder.slots["min"])
+        user.put("checking", user.get("checking") - responder.slots["min"])
+    else:
+        responder.reply(
+            "OK, we have scheduled your credit card payment for your {payment} of $"
+            "{total_balance:.2f}"
+        )
+        user.put("credit", 0)
+        user.put("checking", user.get("checking") - responder.slots["total_balance"])
+
+
+def _exact_amount_helper(request, responder, entity, user):
+    try:
+        responder.slots["amount"] = entity["value"][0]["value"]
+    except KeyError:
+        responder.reply(
+            "Unable to recognize the amount of money you have provided, try "
+            "formatting the value like this '$40.50'"
+        )
+        return
+    if 0 < responder.slots["amount"] <= user.get("credit"):
+        responder.slots["amount"] = entity["value"][0]["value"]
+        responder.reply(
+            "OK, we have scheduled your credit card payment for $"
+            "{amount:.2f}"
+        )
+        user.put("credit", round(user.get("credit") - entity["value"][0]["value"], 2))
+        user.put("checking", round((user.get("checking") - responder.slots["amount"]), 2))
+    else:
+        if responder.slots["amount"] > 0:
```

Comment:
Unnecessary nesting here. Could just do 
```
elif responder.slots["amount"] > 0:
... 
else:
```
It also feels a little weird to me to compare to zero here but then reply about a comparison to the balance.

Reply:
I think the nesting has two cases which suffice the condition for the outermost else.